public/vislib/marker_types/base_marker.js

```diff
@@ -343,10 +349,37 @@ define(function (require) {
           this._legendColors = reds5;
         }
 
-        this._legendQuantizer = d3.scale.quantize().domain(quantizeDomain).range(this._legendColors);
+        if('Dynamic - Linear' === this._attr.scaleType) {
+          this._legendQuantizer = d3.scale.quantize().domain(quantizeDomain).range(this._legendColors);
+        } else { // Dynamic - Uneven
+          // A legend scale that will create uneven ranges for the legend in an attempt
+          // to split the map features uniformly across the ranges.  Useful when data is unevenly
+          // distributed across the minimum - maximum range.
+          let features = this.geoJson.features;
+          features.sort(function(x, y) {
+            return d3.ascending(x.properties.value, y.properties.value);
+          });
+          let featureLength = features.length;
+          let count = 0;
+          let ranges = [];
+          let max = 0;
+          let bands = this._legendColors.length;
+          features.forEach(function(obj) {
+            max = Math.max(max, obj.properties.value);
+            count++;
+            if(count >= featureLength/bands) {
+              count = 0;
+              ranges.push(obj.properties.value);
+            }
+          });
```

Comment:
Is this forEach necessary? Couldn't the ranges just be selected by grabbing the values at the features[index] of range band splits?